src/web/client/common/remoteServiceProvider.ts

```diff
@@ -0,0 +1,147 @@
+    let languageCode;
+    if (languageIdCodeMap.size) {
+        languageCode = languageIdCodeMap.get(queryParamsMap.get('websiteId')) ? languageIdCodeMap.get(queryParamsMap.get('websiteId')) : DEFAULT_LANGUAGE_CODE;
+    }
+    const adxCopy = res['adx_copy'] ? res['adx_copy'] : ' ';
+    const adxCustomJavascript = res['adx_customjavascript'] ? res['adx_customjavascript'] : ' ';
+    const adxCustomCss = res['adx_customcss'] ? res['adx_customcss'] : ' ';
+    const fileName = res['adx_name'] ? res['adx_name'] : 'defaultfilename';
+    portalsFS.createDirectory(vscode.Uri.parse(`${PORTALSURISCHEME}:/${PORTALSFOLDERNAME}/${WEBPAGES_FILENAME}/${fileName}/`, true));
+    portalsFS.createDirectory(vscode.Uri.parse(`${PORTALSURISCHEME}:/${PORTALSFOLDERNAME}/${WEBPAGES_FILENAME}/${fileName}/${CONTENTPAGES}/`, true));
+    createVscodeFile(portalsFS, fileName, languageCode, adxCustomCss, 'customcss.cs');
+    createVscodeFile(portalsFS, fileName, languageCode, adxCustomJavascript, 'customjs.js');
+    createVscodeFile(portalsFS, fileName, languageCode, adxCopy, 'webpage.copy.html');
+    vscode.window.showTextDocument(vscode.Uri.parse(`${PORTALSURISCHEME}:/${PORTALSFOLDERNAME}/${WEBPAGES_FILENAME}/${fileName}/${CONTENTPAGES}/${fileName}.${languageCode}.webpage.copy.html`))
+}
+
+function createVscodeFile(portalsFS: PortalsFS, fileName: string, languageCode: string, data: any, ext: any) {
+    portalsFS.writeFile(vscode.Uri.parse(`${PORTALSURISCHEME}:/${PORTALSFOLDERNAME}/${WEBPAGES_FILENAME}/${fileName}/${CONTENTPAGES}/${fileName}.${languageCode}.${ext}`), new TextEncoder().encode(data), { create: true, overwrite: true });
+}
+
+function createwebtemplate(res: any, entity: string, portalFS: PortalsFS) {
+    portalFS.createDirectory(vscode.Uri.parse(`${PORTALSURISCHEME}:/${PORTALSFOLDERNAME}/${WEBTEMPLATES}/`, true));
+    const adxSource = res['adx_source'] ? res['adx_source'] : 'dummy source';
+    const fileName = entitiesSchemaMap.get(entity)._name;
+    portalFS.createDirectory(vscode.Uri.parse(`${PORTALSURISCHEME}:/${PORTALSFOLDERNAME}/${WEBTEMPLATES}/${fileName}/`, true));
+    portalFS.writeFile(vscode.Uri.parse(`${PORTALSURISCHEME}:/${PORTALSFOLDERNAME}/${WEBTEMPLATES}/${fileName}/${fileName}.${WEBTEMPLATES}.source.html`), new TextEncoder().encode(adxSource), { create: true, overwrite: true });
+}
+
+function createwebfile(res: any, entity: string, portalFS: PortalsFS) {
+    portalFS.createDirectory(vscode.Uri.parse(`${PORTALSURISCHEME}:/${PORTALSFOLDERNAME}/${entity}/`, true));
+    const adxSource = res['adx_filecontent'] ? res['adx_filecontent'] : 'dummy content';
+    const fileName = entitiesSchemaMap.get(entity)._name;
+    portalFS.writeFile(vscode.Uri.parse(`${PORTALSURISCHEME}:/${PORTALSFOLDERNAME}/${entity}/${fileName}.webfile.yml`), new TextEncoder().encode(adxSource), { create: true, overwrite: true });
+}
+
+
+async function saveData(accessToken: string, requestUrl: string, fileUri: vscode.Uri, entity: string, value: string) {
```

Comment:
this is not scalable. How will we handle 'default' app implementation with this?